### PR TITLE
Document create sink as select 

### LINKED ISSUE #453: Document create sink from a select query (create sink as)
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/6648

### Which part(s) of the docs might be affected or should be updated? And how?

Previously we can create sink only from an existing table, view, or mv. Now we can create a sink from a select query (CREATE SINK AS SELECT ...)

### Reference

_No response_

### LINKED ISSUE #503: Bug: sql-create-sink "append-only" should be "append_only"
### Where's the bug?

https://www.risingwave.dev/docs/current/sql-create-sink/

### Describe the bug

 I noticed that there is a typo in `Allowed formats: ‘append-only’`
 This is incorrect because it should be `append_only`. 
 I think you folks should replace `append-only` with `append_only`. 
 If you take a look at https://github.com/risingwavelabs/risingwave/blob/main/src/connector/src/sink/kafka.rs#L71 you will agree with me. 
 Cheers.

### PR DESCRIPTION
**Pull request by hengm3467**
## Info
- **Description**: 
Add descriptions about create sink as select. 

- **Related code PR**: 
[ Provide the link to the code PR here. For example, https://github.com/risingwavelabs/risingwave/pull/4085. Delete this part if there's no code PR related. ]

- **Related doc issue**: 
Resolves #453 

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

File: docs/sql/commands/sql-create-sink.md [hunk 1]
```diff
@@ -6,14 +6,16 @@ slug: /sql-create-sink
...
+You can create a sink from a materialized source, a materialized view, a table, or a `SELECT` query.
 
 
 ## Syntax
 
 ```sql
-CREATE SINK [ IF NOT EXISTS ] sink_name 
-FROM sink_from
+CREATE SINK [ IF NOT EXISTS ] sink_name
+[FROM sink_from | AS SELECT query]
```
Comment (yuhao-su) [inline docs/sql/commands/sql-create-sink.md]: May use `AS select_query` to keep it consistent with create mv doc.
https://github.com/risingwavelabs/risingwave-docs/blob/main/docs/sql/commands/sql-create-mv.md

File: docs/sql/commands/sql-create-sink.md [hunk 2]
```diff
@@ -24,14 +26,38 @@ WITH (
...
+```
+
+```sql
+CREATE SINK sink2 AS SELECT 
+avg(distance) as avg_distance, avg(duration) as avg_duration 
+FROM taxi_trips
+WITH (
+     connector='kafka',
+   kafka.brokers='localhost:9092',
+   kafka.topic='test',
```
Comment (yuhao-su) [inline docs/sql/commands/sql-create-sink.md]:
```suggestion
   connector='kafka',
   kafka.brokers='localhost:9092',
   kafka.topic='test',
```

File: docs/sql/commands/sql-create-sink.md [hunk 3]
```diff
@@ -24,14 +26,38 @@ WITH (
...
 
-All WITH options are required.
 
 |Parameter | Description|
 |---|---|
-|sink_name| Name of the sink to be created|
-|sink_from| The source from which data will be output. It can be a materialized source, a materialized view, or a table.|
+|sink_name| Name of the sink to be created. Required.|
+|FROM sink_from clause| The direct source from which data will be output. It can be a materialized source, a materialized view, or a table. Either this clause or the AS SELECT statement must be specified.|
+|AS SELECT query| A `SELECT` statement to specify the data to be output to the sink. Either this statement or a FROM clause must be specified.|
```
Comment (yuhao-su) [inline docs/sql/commands/sql-create-sink.md]: Ditto

File: docs/sql/commands/sql-create-sink.md (hunk 1 above)
Comment (hengm3467) [reply docs/sql/commands/sql-create-sink.md]: @yuhao-su  Thanks. I updated based on your suggestion.

**Comment by tabVersion** [comment]
please hold the pr for a while. We are going to ban some behavior in `AS SELECT query`.

**Comment by hengm3467** [comment]
> please hold the pr for a while. We are going to ban some behavior in `AS SELECT query`.

OK. Then we don't need to include this one in the 0.1.15 doc branch.

**Comment by hengm3467** [comment]
> please hold the pr for a while. We are going to ban some behavior in `AS SELECT query`.

OK. Then we don't need to include this one in the 0.1.15 doc branch.

**Comment by hengm3467** [comment]
Now we can proceed. @emile-00 Please take a look and make necessary changes. Thanks.

File: docs/sql/commands/sql-create-sink.md [hunk 4]
```diff
@@ -24,14 +26,51 @@ WITH (
...
+## Examples
+
+Create a sink by selecting an entire materialized view.
+```sql
+CREATE SINK sink1 FROM mv1 
+WITH (
+   connector='kafka',
+   kafka.brokers='localhost:9092',
+   kafka.topic='test',
+   format='append-only'
```
Comment (tabVersion) [inline docs/sql/commands/sql-create-sink.md]: append_only

File: docs/sql/commands/sql-create-sink.md [hunk 5]
```diff
@@ -24,14 +26,51 @@ WITH (
...
+)
+```
+
+Create a sink by selecting the average `distance` and `duration` from `taxi_trips`.
```
Comment (tabVersion) [inline docs/sql/commands/sql-create-sink.md:52]: please provide an example of the contents that sink into kafka

**Review by tabVersion — APPROVED** [review]
rest LGTM

File: docs/sql/commands/sql-create-sink.md [hunk 6]
```diff
@@ -24,14 +26,51 @@ WITH (
...
 
-|Parameter | Description|
+|Parameter or clause| Description|
 |---|---|
-|sink_name| Name of the sink to be created|
-|sink_from| The source from which data will be output. It can be a materialized source, a materialized view, or a table.|
+|sink_name| Name of the sink to be created. Required.|
+|FROM sink_from clause| A clause that specifies the direct source from which data will be output. *sink_from* can be a materialized view or a table. Either this clause or a SELECT query must be specified.|
+|AS select_query| A SELECT query that specifies the data to be output to the sink. Either this query or a FROM clause must be specified.See [SELECT](../commands/sql-select.md) for the syntax and examples of the SELECT command.|
 |connector| Sink connector type. Currently, only `‘kafka’` is supported.|
```
Comment (CAJan93) [inline docs/sql/commands/sql-create-sink.md]: Do we have a roadmap which other connector types we want to support in the future? If so, we could link that here.

**Review by tabVersion — APPROVED** [review]

File: docs/sql/commands/sql-create-sink.md (hunk 6 above)
Comment (tabVersion) [reply docs/sql/commands/sql-create-sink.md]: We have this page https://www.risingwave.dev/docs/current/rw-integration-summary/

File: docs/sql/commands/sql-create-sink.md (hunk 5 above)
Comment (hengm3467) [reply docs/sql/commands/sql-create-sink.md:52]: @tabVersion  Did you mean that we need to include the messages that are output to Kafka?

File: docs/sql/commands/sql-create-sink.md (hunk 6 above)
Comment (hengm3467) [reply docs/sql/commands/sql-create-sink.md]: We'll include the 'jdbc' connector in another doc PR and include the link there.

**Pull request merged**